packages/tizen_bundle/lib/tizen_bundle.dart

```diff
@@ -0,0 +1,347 @@
+    int ret = using((Arena arena) {
+      return tizen.bundle_add_str(_handle, key.toNativeInt8(allocator: arena),
+          value.toNativeInt8(allocator: arena));
+    });
+    if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+      Log.error(_logTag,
+          "Failed to add string. key: $key, value: $value, error: $ret");
+      _BundleErrorFactory().throwException(ret);
+    }
+  }
+
+  /// Gets a string from the bundle object with the specific key.
+  String getString(String key) {
+    String value = using((Arena arena) {
+      final pValue = arena<Pointer<Int8>>();
+      int ret = tizen.bundle_get_str(
+          _handle, key.toNativeInt8(allocator: arena), pValue);
+      if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+        Log.error(_logTag, "Failed to get string. key: $key, error: $ret");
+        _BundleErrorFactory().throwException(ret);
+      }
+
+      return pValue.value.toDartString();
+    });
+
+    return value;
+  }
+
+  /// Adds string ino the bundle object.
+  void addStrings(String key, List<String> values) {
+    using((Arena arena) {
+      final pointerList =
+          values.map((str) => str.toNativeInt8(allocator: arena)).toList();
+      final pointerArray = arena<Pointer<Int8>>(pointerList.length);
+      values.asMap().forEach((index, value) {
+        pointerArray[index] = pointerList[index];
+      });
+
+      int ret = tizen.bundle_add_str_array(_handle,
+          key.toNativeInt8(allocator: arena), pointerArray, values.length);
+      if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+        Log.error(
+            _logTag, "Failed to add string array. key: $key, error: $ret");
+        _BundleErrorFactory().throwException(ret);
+      }
+    });
+  }
+
+  /// Gets strings from the bundle object with the specific key.
+  List<String> getStrings(String key) {
+    List<String> values = using((Arena arena) {
+      final arraySize = arena<Int32>();
+      final stringArray = tizen.bundle_get_str_array(
+          _handle, key.toNativeInt8(allocator: arena), arraySize);
+      int ret = tizen.get_last_result();
+      if (ret != bundle_error_e.BUNDLE_ERROR_NONE) {
+        Log.error(
+            _logTag, "Failed to get string array. key: $key, error: $ret");
+        _BundleErrorFactory().throwException(ret);
+      }
+
+      List<String> strings = [];
+      for (int index = 0; index < arraySize.value; ++index) {
+        strings.add(stringArray[index].toDartString());
+      }
+
+      return strings;
+    });
+
+    return values;
+  }
+
+  static void _bundleIteratorCallback(Pointer<Int8> pKey, int type,
+      Pointer<keyval_t> pKeyval, Pointer<Void> userData) {
+    final pCount = userData.cast<Int32>();
+    final String key = pKey.toDartString();
+    _iteratorMap[pCount.value]?.add(KeyInfo(key, type));
+  }
+
+  /// Gets the KeyInfo items from the bundle object.
+  List<KeyInfo>? getKeys() {
+    int count = _iteratorCount++;
+    _iteratorMap[count] = [];
```

Comment:
We don't need to consider a race condition by multi-threading in Dart world. Just use a single Map instance.

Reply:
Done. Thanks.
I've changed it using List<KeyInfo>.